src/modules/vendor/router/routes.js

```diff
@@ -178,6 +178,17 @@ const routes = [
           cookies: ['alenvi_token', 'refresh_token'],
           parent: 'management',
         },
+        beforeEnter: async (_, from, next) => {
```

Comment:
pourquoi ces 2 ajouts ?

Reply:
Pour conserver le filtre sur les formations archivées lorsque :
 - on selectionne un filtre sur le kanban
 - on va sur la fiche d'une formaiotn
 - on revient en arrière

Reply:
En creusant un peu, il n'y a a priori pas de problème d'utiliser l'option api et l'api de composition dans un meme composant. Vue n'envisage pas de rendre obsolète l'api d'option donc je serais partante pour mettre en place ce que tu m'as suggéré à l'oral : utiliser le hook `beforeRouteEnter` directement dans `blendedCourseProfile`

Reply:
Concernant le fait de restorer tous les filtres plutot que de restorer uniquement le filtre sur l'archive, j'ai creusé mais je n'ai pas trouvé de solution concluante. 

Pour utiliser `resetFilters` , il faudrait qu'on ai l'info de l'interface depuis laquelle l'action est appelée. Le soucis est que dans `beforeRouteEnter`, on n'a pas accès a l'instance du composant (`this`). La question est : comment accéder a `isClientInterface` dans ce cas ? 

Une solution que j'ai essayé de mettre en place est de passer une callback a next dans `beforeRouteEnter`. J'ai testé de : 
- creer une donnée `isClientInterface` 
- dispatcher l'action `resetFilters` en lui passant `vm.isClientInterface` 

Ce qui donne: 
```
data () {
    const isClientInterface = !/\/ad\//.test(this.$route.path);

    return {
      isClientInterface,
    };
  },
  beforeRouteEnter (_, from, next) {
    next(vm => {
      console.log('vm', vm.isClientInterface);
      if (from.name !== 'trainers courses info') {
        store.dispatch('course/resetFilters', { isClientInterface: vm.isClientInterface });
      }
    });
  },
```

Reply:
cool pour la cohabitation ! on est pas totalement dans le cas qu'il suggère " we only recommend doing so if you have an existing Options API codebase..." ; nous c'est plutot l'inverse la mais tant qu'il n'y a pas de contre indication je pense qu'on peut partir dessus

et pour le 2nd point en fait tu n'as pas besoin de la variable `isClientInterface` poru savoir ou tu es car tu es dans les fichiers `src/modules/vendor/pages/ni/management/BlendedCoursesDirectory.vue` et `src/modules/vendor/pages/trainers/management/BlendedCoursesDirectory.vue`. C'est l'avantage des fichiers `pages` comme ils ne sont pas mutualisés, tu sais directement ou tu es ;)

Reply:
Ah mais oui 🤦‍♀️